File: TestWAT/Stores/PersonStore.vb
Imports TestWAT.Enums
Imports TestWAT.Models.Filters

Public Class PersonStore

    Public Sub New()
        KeyName = "Key"
        ValueName = "Value"
        FilterColumns = New List(Of String)({"Key", "Value", "Type", "TypeString"})
    End Sub

    Public ReadOnly Property KeyName As String
    Public ReadOnly Property ValueName As String
    Public ReadOnly Property FilterColumns As List(Of String)

    Public Function SelectList() As Task(Of DataTable)
        Return SelectList(New PersonFilter())
    End Function

    Public Async Function SelectList(Filter As PersonFilter) As Task(Of DataTable)
        Dim DT = New DataTable()
        DT.Columns.Add("Key", GetType(Integer))
        DT.Columns.Add("Value", GetType(String))
        DT.Columns.Add("Type", GetType(Integer))
        DT.Columns.Add("TypeString", GetType(String))
        For i = 0 To 150
            Dim Type = CType((i Mod 3) + 1, PersonType)
            If Filter.Type.HasValue AndAlso Type <> Filter.Type Then Continue For

            Dim R = DT.NewRow()
            R("Key") = i
            R("Value") = $"Person {i}"
            R("Type") = CInt(Type)
            R("TypeString") = Type.ToString()
            DT.Rows.Add(R)
        Next

        Await Task.Delay(1000)
        Return DT
    End Function

End Class

File: JANL.Mock/Stores/PersonStore.cs
using JANL.Models.Filters;
using System.Data;
using System.Threading.Tasks;

namespace JANL.Stores
{
    public class PersonStore
    {
        public PersonStore()
        {
            KeyName = "Key";
            ValueName = "Value";
        }

        public string KeyName { get; }
        public string ValueName { get; }

        public Task<DataTable> SelectList() => SelectList(default);

        public async Task<DataTable> SelectList(PersonFilter filter)
        {
            var DT = new DataTable();
            DT.Columns.Add("Key", typeof(int));
            DT.Columns.Add("Value", typeof(string));
            DT.Columns.Add("Type", typeof(int));
            DT.Columns.Add("TypeString", typeof(string));
            for (int i = 0; i < 150; i++)
            {
                var Type = (PersonType)(i % 3) + 1;
                if (filter.Type.HasValue && Type != filter.Type) { continue; }

                var R = DT.NewRow();
                R["Key"] = i;
                R["Value"] = $"Person {i}";
                R["Type"] = (int)Type;
                R["TypeString"] = Type.ToString();
                DT.Rows.Add(R);
            }
            await Task.Delay(1000);
            return DT;
        }
    }
}